### PR TITLE
P3 Weather Async App

### PR DESCRIPTION
**Pull request by leemoispace**
P3 Weather Async App

Rubrics:https://learn-udacity.top/fewd586261/Front%20End%20Web%20Developer%20v5.0.0/Part%2004-Module%2001-Lesson%2005_Project%20Weather%20Journal%20App/Project%20Rubric%20-%20Weather%20Journal%20App.html

video:

https://user-images.githubusercontent.com/2882655/180152256-5c5bf0d7-3286-493d-9430-450a347ecda0.mp4

### CODE REVIEW

File: p3-weather-journal-app/website/app.js [hunk 1]
```diff
@@ -3,3 +3,111 @@
 // Create a new date instance dynamically with JS
```
Comment (Martian2Lee) [inline p3-weather-journal-app/website/app.js]: replace all `let` and `var` with const

File: p3-weather-journal-app/website/app.js [hunk 2]
```diff
@@ -3,3 +3,111 @@
...
+    document.getElementById(
+      "content"
+    ).innerHTML = `Feelings: ${data.userResponse}`;
+  } catch (error) {
+    console.log("error", error);
+  }
+};
+
+// Convert date
+function convertDate(unixtimestamp) {
```
Comment (Martian2Lee) [inline p3-weather-journal-app/website/app.js:40]: if you copied this from else where, please attach the link to the original code

File: p3-weather-journal-app/website/app.js [hunk 3]
```diff
@@ -3,3 +3,111 @@
...
+const getWeatherData = async (baseURL, city, API_KEY) => {
+  let geodata = await getCityCords(cityURL, city, API_KEY);
+  //weather api: https://openweathermap.org/api/one-call-3
+  const url = `${baseURL}lat=${geodata[0].lat}&lon=${geodata[0].lon}&exclude=minutely,hourly,daily,alerts&${API_KEY}`;
+  //const response = await fetch(url);
+  const response = await fetch(url);
+  let data = await response.json();
+  return data;
+};
+//getWeatherData(baseURL,'dubai',API_KEY)
```
Comment (Martian2Lee) [inline p3-weather-journal-app/website/app.js]: remove this

File: p3-weather-journal-app/website/app.js [hunk 4]
```diff
@@ -3,3 +3,111 @@
...
+
+  try {
+    const newData = await response.json();
+    console.log("post res: ", newData);
+  } catch (error) {
+    console.log("error", error);
+  }
+};
+
+//Function to update UI
```
Comment (Martian2Lee) [inline p3-weather-journal-app/website/app.js:67]: // Function to update UI

File: p3-weather-journal-app/website/app.js [hunk 5]
```diff
@@ -3,3 +3,111 @@
...
+      postDataApi("addWeatherData", {
+        temperature: data.current.temp,
+        date: convertDate(data.current.dt),
+        userResponse: feelings,
+      });
+    })
+    .then(() => updateUI());
+}
+
+//Function to POST data
```
Comment (Martian2Lee) [inline p3-weather-journal-app/website/app.js:47]: // Function to POST data

File: p3-weather-journal-app/website/app.js (hunk 1 above)
Comment (Martian2Lee) [inline p3-weather-journal-app/website/app.js]: remove all console log

File: p3-weather-journal-app/website/app.js [hunk 6]
```diff
@@ -3,3 +3,111 @@
...
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(data), // body data type must match "Content-Type" header
+  });
+
+  try {
+    const newData = await response.json();
+    console.log("post res: ", newData);
+  } catch (error) {
+    console.log("error", error);
```
Comment (Martian2Lee) [inline p3-weather-journal-app/website/app.js]: use console.warn in catch, and do console.warn(error.message)

File: p3-weather-journal-app/website/app.js [hunk 7]
```diff
@@ -3,3 +3,111 @@
...
+  //response format as above doc
+  console.log(geodata[0].lat, geodata[0].lon);
+  return geodata;
+};
+
+const getWeatherData = async (baseURL, city, API_KEY) => {
+  let geodata = await getCityCords(cityURL, city, API_KEY);
+  //weather api: https://openweathermap.org/api/one-call-3
+  const url = `${baseURL}lat=${geodata[0].lat}&lon=${geodata[0].lon}&exclude=minutely,hourly,daily,alerts&${API_KEY}`;
+  //const response = await fetch(url);
```
Comment (Martian2Lee) [inline p3-weather-journal-app/website/app.js]: remove this

File: p3-weather-journal-app/website/app.js (hunk 1 above)
Comment (Martian2Lee) [inline p3-weather-journal-app/website/app.js]: comment consistency

File: p3-weather-journal-app/website/app.js [hunk 8]
```diff
@@ -3,3 +3,111 @@
...
+    console.log("updateUI: ", data);
+    document.getElementById("date").innerHTML = `Date: ${data.date}`;
+    document.getElementById(
+      "temp"
+    ).innerHTML = `Temperature(°C): ${data.temperature}`;
+    document.getElementById(
+      "content"
+    ).innerHTML = `Feelings: ${data.userResponse}`;
+  } catch (error) {
+    console.log("error", error);
```
Comment (Martian2Lee) [inline p3-weather-journal-app/website/app.js]: use console.warn in catch, and do console.warn(error.message)

File: p3-weather-journal-app/website/app.js (hunk 1 above)
Comment (Martian2Lee) [inline p3-weather-journal-app/website/app.js]: function order is confusing

**Review by Martian2Lee — CHANGES_REQUESTED** [review]

File: p3-weather-journal-app/website/app.js (hunk 8 above)
Comment (leemoispace) [reply p3-weather-journal-app/website/app.js]: done in: [20ecb96](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/7/commits/20ecb96ba6c5ac532cbc0ea394143deb379eb132)

File: p3-weather-journal-app/website/app.js (hunk 1 above)
Comment (leemoispace) [reply p3-weather-journal-app/website/app.js]: done in: [20ecb96](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/7/commits/20ecb96ba6c5ac532cbc0ea394143deb379eb132)

File: p3-weather-journal-app/website/app.js (hunk 7 above)
Comment (leemoispace) [reply p3-weather-journal-app/website/app.js]: done in: [20ecb96](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/7/commits/20ecb96ba6c5ac532cbc0ea394143deb379eb132)

File: p3-weather-journal-app/website/app.js (hunk 6 above)
Comment (leemoispace) [reply p3-weather-journal-app/website/app.js]: done in: [20ecb96](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/7/commits/20ecb96ba6c5ac532cbc0ea394143deb379eb132)

File: p3-weather-journal-app/website/app.js (hunk 1 above)
Comment (leemoispace) [reply p3-weather-journal-app/website/app.js]: done in: [20ecb96](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/7/commits/20ecb96ba6c5ac532cbc0ea394143deb379eb132)

File: p3-weather-journal-app/website/app.js (hunk 3 above)
Comment (leemoispace) [reply p3-weather-journal-app/website/app.js]: done in: [20ecb96](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/7/commits/20ecb96ba6c5ac532cbc0ea394143deb379eb132)

File: p3-weather-journal-app/website/app.js (hunk 2 above)
Comment (leemoispace) [reply p3-weather-journal-app/website/app.js:40]: done in: [20ecb96](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/7/commits/20ecb96ba6c5ac532cbc0ea394143deb379eb132)

File: p3-weather-journal-app/website/app.js (hunk 1 above)
Comment (leemoispace) [reply p3-weather-journal-app/website/app.js]: done in: [20ecb96](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/7/commits/20ecb96ba6c5ac532cbc0ea394143deb379eb132)

Comment (leemoispace) [reply p3-weather-journal-app/website/app.js]: done in: [0648763](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/7/commits/064876345af95540b541656b9526dcd75c2822f0)

**Review by Martian2Lee — APPROVED** [review]
🚀

**Pull request merged**